### ISSUE TITLE
CVC4 not responding "success" immediately after define-fun

### ISSUE THREAD

**Issue by 4tXJ7f** [CVC4/CVC4#1113]
Daniel reported that CVC4 is blocking kind2 by not responding with "success" after certain commands/sequences. I am attaching the example input that he sent me. I was able to reproduce the issue with the following Python script:

```
import subprocess

with open('traffic_e7_46.lus.ic3.1.smt2') as f:
        lines = f.readlines()

p = subprocess.Popen(["builds/bin/cvc4", "--lang=smt2"], stdout=subprocess.PIPE, stdin=subprocess.PIPE, bufsize=1)

buf = ''
for line in lines:
        if line == ';; Success \n':
                print >>p.stdin, buf
                p.stdin.flush()
                print p.stdout.readline()
                buf = ''
        else:
                buf += line
```

The script blocks after sending the first define-fun command. According to Daniel, the issue does not appear with version 1.5, so I will to a bisect to find out where things started to go wrong.

[Block_Issue.tar.gz](https://github.com/CVC4/CVC4/files/1313269/Block_Issue.tar.gz)

**Comment by 4tXJ7f** [comment]
The issue was introduced in commit 6d82ee2d1f84065ff4a86f688a3b671b85728f80, likely due to the change of how `let` statements are parsed.

**Comment by 4tXJ7f** [comment]
Thanks for the explanations! Yes, I've narrowed it down to the let definition as well. What is really strange is that either adding a space after the `let` or removing the new line after `let` seems to fix the issue as well. My guess would be that it is either an issue with ANTLR or (maybe more likely) an issue with our buffering code when reading from stdin. I am going to take a closer look at the issue tomorrow.

**Comment by 4tXJ7f** [comment]
Fixed in 2567868bdf5664e95f998e76512c84d2c63f09a2.
